tests/simplex/step-38.cc

```diff
@@ -0,0 +1,445 @@
+  };
+
+  template <>
+  double
+  Solution<2>::value(const Point<2> &p, const unsigned int) const
+  {
+    return (-2. * p(0) * p(1));
+  }
+
+  template <>
+  Tensor<1, 2>
+  Solution<2>::gradient(const Point<2> &p, const unsigned int) const
+  {
+    Tensor<1, 2> return_value;
+    return_value[0] = -2. * p(1) * (1 - 2. * p(0) * p(0));
+    return_value[1] = -2. * p(0) * (1 - 2. * p(1) * p(1));
+
+    return return_value;
+  }
+
+  template <>
+  double
+  Solution<3>::value(const Point<3> &p, const unsigned int) const
+  {
+    return (std::sin(numbers::PI * p(0)) * std::cos(numbers::PI * p(1)) *
+            exp(p(2)));
+  }
+
+  template <>
+  Tensor<1, 3>
+  Solution<3>::gradient(const Point<3> &p, const unsigned int) const
+  {
+    using numbers::PI;
+
+    Tensor<1, 3> return_value;
+
+    return_value[0] = PI * cos(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    return_value[1] = -PI * sin(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+    return_value[2] = sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+
+    return return_value;
+  }
+
+  template <int dim>
+  class RightHandSide : public Function<dim>
+  {
+  public:
+    virtual double
+    value(const Point<dim> &p, const unsigned int component = 0) const override;
+  };
+
+  template <>
+  double
+  RightHandSide<2>::value(const Point<2> &p,
+                          const unsigned int /*component*/) const
+  {
+    return (-8. * p(0) * p(1));
+  }
+
+  template <>
+  double
+  RightHandSide<3>::value(const Point<3> &p,
+                          const unsigned int /*component*/) const
+  {
+    using numbers::PI;
+
+    Tensor<2, 3> hessian;
+
+    hessian[0][0] = -PI * PI * sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    hessian[1][1] = -PI * PI * sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    hessian[2][2] = sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+
+    hessian[0][1] = -PI * PI * cos(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+    hessian[1][0] = -PI * PI * cos(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+
+    hessian[0][2] = PI * cos(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    hessian[2][0] = PI * cos(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+
+    hessian[1][2] = -PI * sin(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+    hessian[2][1] = -PI * sin(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+
+    Tensor<1, 3> gradient;
+    gradient[0] = PI * cos(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    gradient[1] = -PI * sin(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+    gradient[2] = sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+
+    Point<3> normal = p;
+    normal /= p.norm();
+
+    return (-trace(hessian) + 2 * (gradient * normal) +
+            (hessian * normal) * normal);
+  }
+
+  template <int spacedim>
+  LaplaceBeltramiProblem<spacedim>::LaplaceBeltramiProblem(
+    const unsigned degree)
+    : dof_handler(triangulation)
+    , fe(degree)
+#ifdef USE_SIMPLEX
+    // Quadratic mapping for simplex is still not implemented.
+    , mapping(Simplex::FE_P<dim, spacedim>(degree))
```

Comment:
We are working on that! See #11563.

Reply:
That's nice. I will try to test quadratic when #11563 is ready to get merged.

Reply:
@lwy5515 It's on master :)